src/main/java/com/celements/nextfreedoc/NextFreeDocService.java

```diff
@@ -17,22 +17,28 @@
 import org.xwiki.query.QueryException;
 import org.xwiki.query.QueryManager;
 
+import com.celements.model.access.IModelAccessFacade;
+import com.celements.model.access.exception.DocumentAlreadyExistsException;
+import com.celements.model.access.exception.DocumentLoadException;
 import com.xpn.xwiki.XWikiContext;
 import com.xpn.xwiki.XWikiException;
 
 @Singleton
 @Component
 public class NextFreeDocService implements INextFreeDocRole {
 
-  private static Log LOGGER = LogFactory.getFactory().getInstance(NextFreeDocService.class);
+  private static Logger LOGGER = LoggerFactory.getLogger(NextFreeDocService.class);
 
   // TODO refactor to org.xwiki.cache.CacheManager
-  private final Map<DocumentReference, Long> numCache = new HashMap<DocumentReference, Long>();
+  private final Map<DocumentReference, Long> numCache = new ConcurrentHashMap<DocumentReference, Long>();
```

Comment:
HashMap is fine, all (non-test) access to the map is contained in a synchronized block